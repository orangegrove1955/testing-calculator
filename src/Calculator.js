import React, { useState } from "react";

export default function Calculator() {
  const [displayValue, setDisplayValue] = useState(0);
  const [numbers, setNumbers] = useState([]);
  const [operators, setOperators] = useState([]);
  const [selectedOperator, setSelectedOperator] = useState("");
  const [storedValue, setStoredValue] = useState("");

  const callOperator = () => {
    console.log("call operation");
  };

  const setOperator = () => {
    console.log("set operation");
  };

  const updateDisplay = () => {
    console.log("update display");
  };

  return <div className="calculator-container"></div>;
}
