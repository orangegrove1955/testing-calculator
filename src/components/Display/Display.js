import React from "react";
import PropTypes from "prop-types";

const Display = ({ displayValue }) => {
  return (
    <div className="display-container">
      <p className="display-value">{displayValue}</p>
    </div>
  );
};

Display.propTypes = { displayValue: PropTypes.number.isRequired };

export default Display;
