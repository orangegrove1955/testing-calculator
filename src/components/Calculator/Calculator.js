import React, { useState } from "react";
import "./Calculator.css";
import Display from "../Display/Display";
import Keypad from "../Keypad/Keypad";

const Calculator = () => {
  const [displayValue, setDisplayValue] = useState(0);
  const [numbers, setNumbers] = useState([]);
  const [operators, setOperators] = useState([]);
  const [selectedOperator, setSelectedOperator] = useState("");
  const [storedValue, setStoredValue] = useState("");

  const callOperator = () => {
    console.log("call operation");
  };

  const setOperator = () => {
    console.log("set operation");
  };

  const updateDisplay = () => {
    console.log("update display");
  };

  return (
    <div className="calculator-container">
      <Display displayValue={displayValue} />
      <Keypad />
    </div>
  );
};

export default Calculator;
