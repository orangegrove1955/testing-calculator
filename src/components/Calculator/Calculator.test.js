import React from "react";
import { shallow } from "enzyme";
import Calculator from "./Calculator";
import Display from "../Display/Display";
import Keypad from "../Keypad/Keypad";

describe("Calculator component", () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<Calculator />)));

  it("should match Calculator snapshot render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render a Display component", () => {
    // Mock value to pass into displayValue
    const displayMock = jest.fn();
    const displayValue = displayMock.mockReturnValueOnce(0);

    expect(
      wrapper.containsMatchingElement(<Display displayValue={displayValue()} />)
    ).toEqual(true);
  });

  it("should render a Keypad component", () => {
    // Mock values to pass to into Keypad
    const keypadMock = jest.fn();
    const keypadValues = keypadMock.mockReturnValueOnce();

    expect(wrapper.containsMatchingElement(<Keypad />)).toEqual(true);
  });
});
