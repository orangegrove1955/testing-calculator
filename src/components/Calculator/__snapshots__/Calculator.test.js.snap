// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should match Calculator snapshot render 1`] = `ShallowWrapper {}`;
